modules/gdscript/gdscript_parser.cpp

```diff
@@ -3644,16 +3635,55 @@ bool GDScriptParser::export_annotations(const AnnotationNode *p_annotation, Node
 	variable->export_info.type = t_type;
 	variable->export_info.hint = t_hint;
 
+	const int64_t max_flags = 32;
+
+	if (p_annotation->name == SNAME("@export_flags") && p_annotation->resolved_arguments.size() > max_flags) {
+		push_error(vformat(R"(The argument count limit for "@export_flags" is exceeded (%d/%d).)", p_annotation->resolved_arguments.size(), max_flags), p_annotation);
+		return false;
+	}
+
 	String hint_string;
 	for (int i = 0; i < p_annotation->resolved_arguments.size(); i++) {
-		if (p_annotation->name != SNAME("@export_placeholder") && String(p_annotation->resolved_arguments[i]).contains(",")) {
-			push_error(vformat(R"(Argument %d of annotation "%s" contains a comma. Use separate arguments instead.)", i + 1, p_annotation->name), p_annotation->arguments[i]);
-			return false;
+		String arg_string = String(p_annotation->resolved_arguments[i]);
+
+		if (p_annotation->name != SNAME("@export_placeholder")) {
+			if (arg_string.is_empty()) {
+				push_error(vformat(R"(Argument %d of annotation "%s" is empty.)", i + 1, p_annotation->name), p_annotation->arguments[i]);
+				return false;
+			}
+			if (arg_string.contains(",")) {
+				push_error(vformat(R"(Argument %d of annotation "%s" contains a comma. Use separate arguments instead.)", i + 1, p_annotation->name), p_annotation->arguments[i]);
+				return false;
+			}
 		}
+
+		if (p_annotation->name == SNAME("@export_flags")) {
+			Vector<String> t = arg_string.split(":", true, 1);
+			if (t[0].is_empty()) {
+				push_error(vformat(R"(Invalid argument %d of annotation "@export_flags": Expected flag name.)", i + 1), p_annotation->arguments[i]);
+				return false;
+			}
+			if (t.size() == 2) {
+				if (t[1].is_empty()) {
+					push_error(vformat(R"(Invalid argument %d of annotation "@export_flags": Expected flag value.)", i + 1), p_annotation->arguments[i]);
+					return false;
+				}
+				if (!t[1].is_valid_int()) {
+					push_error(vformat(R"(Invalid argument %d of annotation "@export_flags": The flag value must be a valid integer.)", i + 1), p_annotation->arguments[i]);
+					return false;
+				}
+				int64_t value = t[1].to_int();
+				if (value < 1 || value >= (1LL << max_flags)) {
```

Comment:
```c++
const int64_t max_flags = 32;
int64_t value = t[1].to_int();

value >= (1LL << max_flags)
```

I'm not sure if this is correct.

Reply:
`1LL << max_flags` should be `4294967296` which is correct. The `>=` comparison is also correct, since the value needs to be less than the maximum; otherwise, it would overflow and become 0.

Reply:
I'm not sure if `int_64t` is needed for the constant `32`, but I haven't found another way to satisfy CI (tests on multiple platforms have failed).

Reply:
The value `1 << 32` doesn't fit in an 32-bit integer. So I'd say it needs to be stored in a `int64_t`. As for the `max_flags` constant, I don't think it needs to be `int64_t`, unless the compiler is complaining that mixing types in bitwise operations.

Reply:
The 2^32 value would still not fit in a 32-bit unsigned integer. It will fit in both a signed and unsigned 64-bit integer. I don't know if it matters in C++ if the 64-bit integer is signed, but yes flags usually are unsigned.